File: 6.2-SparseAndPolyynomial/programs/python/program1.py
class Element:
  def __init__(self, i, j, x):
    self.i = i
    self.j = j
    self.x = x

class Sparse:
  def __init__(self):
    self.ele = []

  def create(self):
    self.m, self.n = map(int, input("Enter dimensions: ").split())
    self.num = int(input("Enter number of non-zero elements: "))
    print("Enter non-zero elements: ")
    for _ in range(self.num):
      i, j, x = map(int, input().split())
      self.ele.append(Element(i, j, x))

  def display(self):
    k = 0
    for i in range(self.m):
      for j in range(self.n):
        if k < self.num and i == self.ele[k].i and j == self.ele[k].j:
          print(self.ele[k].x, end=" ")
          k += 1
        else:
          print("0", end=" ")
      print()

s = Sparse()
s.create()
s.display()

File: 6.2-SparseAndPolyynomial/programs/ruby/program1.rb
class Element
  attr_accessor :i, :j, :x

  def initialize(i, j, x)
      @i = i
      @j = j
      @x = x
  end
end

class Sparse
  attr_accessor :m, :n, :num, :ele

  def initialize
      @ele = []
  end

  def create
      print "Enter dimensions: "
      @m, @n = gets.split.map(&:to_i)
      print "Enter number of non-zero elements: "
      @num = gets.to_i
      print "Enter non-zero elements: "
      @num.times do
          i, j, x = gets.split.map(&:to_i)
          @ele << Element.new(i, j, x)
      end
      puts " @ele: #{@ele}"
      puts " @m: #{@m}"
      puts " @n: #{@n}"
  end

  def display
      k = 0
      (0...@m).each do |i|
          (0...@n).each do |j|
              if k < @num && i == @ele[k].i && j == @ele[k].j
                  print "#{@ele[k].x} "
                  k += 1
              else
                  print "0 "
              end
          end
          puts
      end
  end
end

s = Sparse.new
s.create
s.display
